### DISCUSSION TITLE
Using new Maven build cache?

### DISCUSSION THREAD

**Discussion by vogella** [eclipse-tycho/tycho#2076] [Ideas]
Can Tycho benefit from the new maven build cache extension? Are their any plans already for doing so?

https://maven.apache.org/extensions/maven-build-cache-extension/getting-started.html

**Comment by laeubi** [comment]
According from the doc, buildcache "works" by enabling it in the `.mvn/extensions.xml` and add some configuration stuff,. I didn't find anything that that plugins can do here, but its quite likley that build-cache is only written with standard maven plugins in mind not other extensions.

I once opened a ticket here:
- https://github.com/eclipse-tycho/tycho/issues/587

but it got literally zero interest so far ...

**Comment by lppedd** [comment]
Again this screws up IDEs that actually work with the Maven model lol

> The cache doesn't restore full project state

> In order to work efficiently, IDE should be configured to not use Maven output (target) directories for compilation

This is worth using only on CI builds apparently. But at that point, it probably matters only if you have a gigantic workspace.  
Maven Daemon seems to be the tool to prefer anyway.
